File: apps/api/app/schemas/onboarding.py
from typing import Literal

from pydantic import BaseModel


class OnboardingStepOut(BaseModel):
    key: str
    label: str
    description: str
    kind: Literal["main", "recommended", "milestone"]
    completed: bool
    action_label: str
    action_href: str


class OnboardingOut(BaseModel):
    completed: bool
    main_configuration_complete: bool
    completed_steps: int
    total_steps: int
    steps: list[OnboardingStepOut]

File: apps/web/src/lib/onboarding.ts
export type OnboardingStep = {
  key: string;
  label: string;
  description: string;
  kind: "main" | "recommended" | "milestone";
  completed: boolean;
  action_label: string;
  action_href: string;
};

export type OnboardingProgress = {
  completed: boolean;
  main_configuration_complete: boolean;
  completed_steps: number;
  total_steps: number;
  steps: OnboardingStep[];
};
